gensim/models/word2vec.py

```diff
@@ -1186,12 +1226,56 @@ def scan_vocab(self, sentences, progress_per=10000, trim_rule=None):
                 utils.prune_vocab(vocab, min_reduce, trim_rule=trim_rule)
                 min_reduce += 1
 
+        corpus_count = sentence_no + 1
+        self.raw_vocab = vocab
+        return total_words, corpus_count
+
+    def _scan_vocab_multistream(self, input_streams, workers, trim_rule):
+        manager = multiprocessing.Manager()
+        progress_queue = manager.Queue()
+
+        logger.info("Scanning vocab in %i processes.", min(workers, len(input_streams)))
+        pool = multiprocessing.Pool(processes=min(workers, len(input_streams)))
+
+        results = [
+            pool.apply_async(_scan_vocab_worker,
+                             (stream, progress_queue, self.max_vocab_size, trim_rule)
+                             ) for stream in input_streams
+        ]
+        pool.close()
+
+        unfinished_tasks = len(results)
+        total_words = 0
+        sentence_no = -1
```

Comment:
The semantics might be clearer here if this was called `total_sentences` and initialized to 0. Unless I'm misunderstanding something

Reply:
Agree